src/Arachne.cc

```diff
@@ -767,28 +815,30 @@ signal(ThreadId id) {
     // state, and retry the CAS if it is not. This approach avoids first taking
     // a cache miss to read and then performing a CAS in the case when the
     // target thread is actually BLOCKED.
-    // Experiments show that this approach can save 40 ns compared to explicitly reading
-    // wakeupTimeInCycles and then performing the CAS.
+    // Experiments show that this approach can save 40 ns compared to explicitly
+    // reading wakeupTimeInCycles and then performing the CAS.
     //
     // When explicitly reading, we pay:
     //     1 cache miss to read on the signaler core.
     //     1 cache invalidate other caches during the CAS.
     //     1 cache miss to read on the target core.
-    // Under the CAS-first approach, if the wakeupTimeInCycles == BLOCKED, we pay:
+    // Under the CAS-first approach, if the wakeupTimeInCycles == BLOCKED, we
+    // pay:
     //     1 cache invalidate other caches during the CAS.
     //     1 cache miss to read on the target core.
     uint64_t oldWakeupTime = ThreadContext::BLOCKED;
     uint64_t newValue = 0L;
-    oldWakeupTime = compareExchange(&id.context->wakeupTimeInCycles, oldWakeupTime, newValue);
+    oldWakeupTime = compareExchange(&id.context->wakeupTimeInCycles,
+                                    oldWakeupTime, newValue);
```

Comment:
We have also discussed this in the past; as a reminder, we CAS to avoid accidentally signalling a thread that just exited, which might cause us to attempt to execute on an empty `ThreadContext`.